run_kraken.py

```diff
@@ -491,19 +502,102 @@ def main(cfg):
         help="output report location",
         default="kraken.report",
     )
-    
+
+
+
+    parser.add_option(
+        "--junit-testcase",
+        dest="junit_testcase",
+        help="junit test case description",
+        default=None,
+    )
+
+    parser.add_option(
+        "--junit-testcase-path",
+        dest="junit_testcase_path",
+        help="junit test case path",
+        default=None,
+    )
+
+    parser.add_option(
+        "--junit-testcase-version",
+        dest="junit_testcase_version",
+        help="junit test case version",
+        default=None,
+    )
+
     (options, args) = parser.parse_args()
     report_file = options.output
+    tee_handler = TeeLogHandler()
+    handlers = [logging.FileHandler(report_file, mode="w"), logging.StreamHandler(), tee_handler]
+
     logging.basicConfig(
         level=logging.INFO,
         format="%(asctime)s [%(levelname)s] %(message)s",
-        handlers=[
-            logging.FileHandler(report_file, mode="w"),
-            logging.StreamHandler(),
-        ],
+        handlers=handlers,
     )
+    option_error = False
+
+    # used to check if there is any missing or wrong parameter that prevents
+    # the creation of the junit file
+    junit_error = False
+    junit_normalized_path = None
+    retval = 0
+    junit_start_time = time.time()
+    # checks if both mandatory options for junit are set
+    if options.junit_testcase_path and not options.junit_testcase:
+        logging.error("please set junit test case description with --junit-testcase [description] option")
+        option_error = True
+        junit_error = True
+
+    if options.junit_testcase and not options.junit_testcase_path:
+        logging.error("please set junit test case path with --junit-testcase-path [path] option")
+        option_error = True
+        junit_error = True
+
+    # normalized path
+    if options.junit_testcase:
+        junit_normalized_path = os.path.normpath(options.junit_testcase_path)
+
+        if not os.path.exists(junit_normalized_path):
+            logging.error(f"{junit_normalized_path} do not exists, please select a valid path")
+            option_error = True
+            junit_error = True
+
+        if not os.path.isdir(junit_normalized_path):
+            logging.error(f"{junit_normalized_path} is a file, please select a valid folder path")
+            option_error = True
+            junit_error = True
+
+        if not os.access(junit_normalized_path, os.W_OK):
+            logging.error(f"{junit_normalized_path} is not writable, please select a valid path")
+            option_error = True
+            junit_error = True
+
     if options.cfg is None:
         logging.error("Please check if you have passed the config")
-        sys.exit(1)
+        option_error = True
+
+    if option_error:
+        retval = 1
     else:
-        main(options.cfg)
+        retval = main(options.cfg)
```

Comment:
We should improve the error handling at all levels in order to centralize the return values in this place. Mainly we should remove all the `sys.exit(x)` in the codebase and replace them with exception trying to catch them in a single place and returning the correct exit status from in the main function, this might become a new issue.

Reply:
Let's create a issue to track it @tsebastiani